mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulEditProfileDialog.kt

```diff
@@ -0,0 +1,62 @@
+package ch.epfl.sdp.mobile.state
+
+import androidx.compose.runtime.*
+import ch.epfl.sdp.mobile.application.authentication.AuthenticatedUser
+import ch.epfl.sdp.mobile.ui.setting.EditProfileNameDialog
+import ch.epfl.sdp.mobile.ui.setting.EditProfileNameDialogState
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+/**
+ * Implementation of the [EditProfileNameDialogState] interface
+ *
+ * @param user the current [AuthenticatedUser].
+ * @param scope the coroutine scope.
+ * @param onSave the callback called after we saved the changes.
+ * @param onCancle the callback called when we click the cancle button.
+ */
+class EditProfileNameDialogStateImpl(
+    user: AuthenticatedUser,
+    private val scope: CoroutineScope,
+    onSaveAction: State<() -> Unit>,
+    onCancleAction: State<() -> Unit>
```

Comment:
Are these really different ? Shouldn't we only have an `closeDialog: State<() -> Unit>`-like parameter instead ?

Reply:
No they are exactly the same. However, I want to make the code more readable and thus I separated them.

Reply:
So it is similar to the other dialog we have in `mobile/src/main/java/ch/epfl/sdp/mobile/ui/prepare_game/PrepareGameScreenState.kt`